src/content/docs/en/pages/guides/marketplace/integrations/rate-limiting-with-penalty/rate-limiting-with-penalty.mdx

```diff
@@ -0,0 +1,192 @@
+| `rate_limit_repenalize` | If `true`, the penalty time is recalculated every time a penalized user makes a request. |
+| `rate_limits` | It defines the rate limit windows as an object. You must **add at least one** object. |
+| `algorithm` | It defines the rate limit algorithm to be applied. Possible values: `fixed_window`, `sliding_window`, `token_bucket`. [Read more](#important). |
+| `requests` | Maximum number of requests until reaching the rate limit. |
+| `interval` | Rate limit window time interval. This variable follows the Upstash standard: `XXXX y`, where the X are the numeral and y the unit of measurement, which can be `s` for seconds or `m` for minutes. Example: `120 s`. |
+| `start and end`| These variables define the time window. It uses the 24-hour format and the UTC time zone. [Read more](#important). |
+| `penalty_in_seconds` | Penalty time (blocking that retunrs `403`status code) that users will be subjected to after violating the rate limit. [Read more](#important) |
+
+:::note
+The key to a rate limit is made up of the combination of all these variables. In this code sample, the final key would be:
+`my_rate_limit + User IP + X-a-custom-reader Header Value + Hostname used in the request`.
+
+Example:
+`my_rate_limit_127.0.0.1_Teste_azion.com`.
+:::
+
+#### Important
+
+- This integration permits you to define different rate limit windows for different periods of the day, according to your needs. Example: you can define a limit of `10 requests/minute` from `00:00` to `12:00`, and `15 requests/minute` from `12:01` to `23:59`.
+
+- This integration supports three possible values for `algorithm`:
+  - `fixed_window` divides time into fixed durations/windows.
+  - `sliding_window`builds on top of fixed window but uses a rolling window. Example: you want to define a rate limit of 10 requests per 1 minute, then, you divide the time into 1 minute slices, like in the `fixed window` algorithm.
+  - `token_bucket` defines the maximum number of tokens to fill a bucket and the interval which the bucket will be cleaned. Every request removes one token and if there is no token to take, the request is rejected.
+
+- The `token_bucket` includes different values, where:
+  - `max_tokens` defines the number of tokens (or keys) that this rate limit will allow.
+  - `refil_rate` defines the number of buckets that will be "cleaned" at each time interval.
+
+```bash
+"algorithm": "token_bucket",
+  	    "refil_rate": 5,
+  	    "max_tokens": 5,
+  	    "interval": "10 s",
+           "start": "23:01",
+            "penalty_in_seconds": 55
+```
+
+In this example, every `10 seconds`, `5 buckets` are cleaned, out of a maximum of `5 "busy" buckets`.
+
+- If `start` isn't defined, the default value will be `00:00`. If `end` isn't defined, the default value will be `23:59`" will be assumed. In case of overlaps between rate limits, the rate limit that is first in the JSON Args will be considered.
```

Comment:
```suggestion
- If `start` isn't defined, the default value will be `00:00`. If `end` isn't defined, the default value of `23:59`" will be assumed. In case of overlaps between rate limits, the rate limit that is first in the JSON Args will be considered.
```